summingbird-batch/src/main/scala/com/twitter/summingbird/batch/Batcher.scala

```diff
@@ -169,7 +169,13 @@ trait Batcher extends Serializable {
     dateToBatch(interval)(truncateUp)(truncateDown)
 
   def toInterval(b: BatchID): Interval[Timestamp] =
-    Intersection(InclusiveLower(earliestTimeOf(b)), ExclusiveUpper(earliestTimeOf(b.next)))
+    if (b.id < Long.MaxValue) {
```

Comment:
Can we use `batchIdSuccessible.next(b)` then?
If `earliestTimeOf` will return an `Option` it would be possible to `map` `next` with `earliestTimeOf` to make this function correct.